nomad/drainer/watch_nodes.go

```diff
@@ -179,21 +174,31 @@ func (w *nodeDrainWatcher) watch() {
 			currentNode, tracked := tracked[nodeID]
 
 			switch {
-			// If the node is tracked but not draining, untrack
+
 			case tracked && !newDraining:
+				// If the node is tracked but not draining, untrack
 				w.tracker.Remove(nodeID)
 
-				// If the node is not being tracked but is draining, track
+			case !tracked && node.UnresponsiveStatus():
+				// Down or Disconnected nodes can't be migrated, so skip
+				// tracking them
+
 			case !tracked && newDraining:
+				// If the node is not being tracked but is draining, track
 				w.tracker.Update(node)
 
-				// If the node is being tracked but has changed, update:
+			case tracked && node.UnresponsiveStatus():
+				// Down or Disconnected nodes can't be migrated, so stop
```

Comment:
What about `unknown` allocs on `disconnected` nodes? Will something else ensure they're marked with DesiredStatus=stop if they reconnect?

Reply:
(Great job pointing out a `down` node can come back before we reschedule its work! Distributed systems :flushed:)